conan/internal/deploy.py

```diff
@@ -66,10 +65,22 @@ def full_deploy(graph, output_folder):
             folder_name = os.path.join(folder_name, build_type)
         if arch:
             folder_name = os.path.join(folder_name, arch)
-        new_folder = os.path.join(output_folder, folder_name)
-        rmdir(new_folder)
-        shutil.copytree(dep.package_folder, new_folder, symlinks=True)
-        dep.set_deploy_folder(new_folder)
+        _deploy_single(dep, conanfile, output_folder, folder_name)
+
+
+def _deploy_single(dep, conanfile, output_folder, folder_name):
+    new_folder = os.path.join(output_folder, folder_name)
+    rmdir(new_folder)
+    symlinks = conanfile.conf.get("tools.deployer:symlinks", check_type=bool, default=True)
+    try:
+        shutil.copytree(dep.package_folder, new_folder, symlinks=symlinks)
+    except Exception as e:
```

Comment:
This is quite difficult to test, to make it fail we could be mocking ``shutil.copytree`` because it is not possible for a test to force the symlinks error as that requires multi-platform